pkg/cmdutil/repo_override.go

```diff
@@ -68,3 +70,34 @@ func OverrideBaseRepoFunc(f *Factory, override string) func() (ghrepo.Interface,
 	}
 	return f.BaseRepo
 }
+
+func PromptForRepo(baseRepo ghrepo.Interface, remotes func() (ghContext.Remotes, error), survey prompter.Prompter) (ghrepo.Interface, error) {
+	var defaultRepo string
+	var remoteArray []string
+
+	if remotes, _ := remotes(); remotes != nil {
+		if defaultRemote, _ := remotes.ResolvedRemote(); defaultRemote != nil {
+			// this is a remote explicitly chosen via `repo set-default`
+			defaultRepo = ghrepo.FullName(defaultRemote)
+		} else if len(remotes) > 0 {
+			// as a fallback, just pick the first remote
+			defaultRepo = ghrepo.FullName(remotes[0])
+		}
+
+		for _, remote := range remotes {
+			remoteArray = append(remoteArray, ghrepo.FullName(remote))
+		}
+	}
+
+	baseRepoInput, errInput := survey.Select("Select a base repo", defaultRepo, remoteArray)
+	if errInput != nil {
+		return baseRepo, errInput
+	}
```

Comment:
Everything from this logic tracks for me.  My main question is whether situations where `gh repo set-default` has explicitly set a remote / base should just use it rather than falling back to the prompting logic 🤔

Reply:
I was in doubt here, it would kind of feel like forced magic here. Yes, you might set the default but there could be another repo you want to create secrets on, then you need to remember to override with the `--repo` flag because there would be no way in interactive mode to switch repos once you have a `set-default` one. I thought being explicit and transparent to the user would be the safer bet here

Reply:
oooh I see, thanks for explaining it to me, I should've seen it in the docs. Even if it would be counter intuitive that would be for another discussion. Given the guidelines of `set-default` I completely agree we can just take the default remote and use that.
PR updated, I'll look into add tests soon!